src/main/jbake/content/blog/2017/07/jaeger-with-security-proxy.adoc

```diff
@@ -0,0 +1,252 @@
+the authentication and authorization constraints.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-18-teaser.png[Jaeger login screen]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-07-18-teaser.png[Jaeger login screen]
+endif::[]
+
+To demonstrate that, we'll make use of link:http://keycloak.org[Keycloak] as our security solution, but
+the idea can be adapted to work with any security proxy. This demo should also work without changes with
+link:https://access.redhat.com/products/red-hat-single-sign-on[Red Hat SSO]. For this exercise, we'll need:
+
+* A Keycloak (or Red Hat SSO) server instance running. We'll call its location `${REDHAT_SSO_URL}`
+* An OpenShift cluster, where we'll run Jaeger backend components. It might be as easy as `oc cluster up`
+* A local clone of the link:https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml[Jaeger OpenShift Production template]
+
+Note that we are not trying to secure the communication between the components, like from the Agent to the Collector.
+For this scenario, there are other techniques that can be used, such as mutual authentication via certificates, employing
+link:https://istio.io/[istio] or other similar tools.
+
+=== Preparing Keycloak
+
+For this demo, we'll run Keycloak via Docker directly on the host machine. This is to stress that Keycloak
+does *not* need to be running on the same OpenShift cluster as our Jaeger backend. 
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-18-keycloak.png[Keycloak login screen]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-07-18-keycloak.png[Keycloak login screen]
+endif::[]
+
+The following command should start an appropriate Keycloak server locally. If you already have your
+own Keycloak or Red Hat SSO server, skip this step.
+
+```bash
+docker run --rm --name keycloak-server -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=password -p 8080:8080 jboss/keycloak
+```
+
+Once the Keycloak server is up and running, let's create a realm for Jaeger:
+
+* Login into Keycloak (http://<YOUR_IP>:8080/auth/admin/master/console) with `admin` as username and `password` as password
+* On the top left corner, mouse over the `Select realm` box and click `Add realm`. Name it `jaeger` and click `Create`
+* On `Clients`, click `Create` and set `proxy-jaeger` as the name and save it
+* Set the `Access Type` to `confidential` and `*` as `Valid Redirect URIs` and save it. You might want to fine tune this in a 
+production environment, otherwise you might be open to an attack known as 
+link:https://www.owasp.org/index.php/Unvalidated_Redirects_and_Forwards_Cheat_Sheet["Unvalidated Redirects and Forwards"].
+* Open the `Installation` tab and select `Keycloak OIDC JSON` and copy the JSON that is shown. It should look like this:
+
+```json
+{
+  "realm": "jaeger",
+  "auth-server-url": "http://192.168.2.111:8080/auth",
```

Comment:
Probably should atleast have a note at the end of the snippet to say how to set a value relevant for their environment?